src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+    this.setAnimal();
+  }
+
+  setAnimal() {
+    Animal.alive = [...Animal.alive, this];
+  }
+
+  removeAnimal(animal) {
+    Animal.alive = Animal.alive.filter((el) => el !== animal);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    if (!herbivore.hidden && herbivore instanceof Herbivore) {
+      herbivore.health -= 50;
+
+      if (herbivore.health <= 0) {
+        super.removeAnimal(herbivore);
```

Comment:
The method removeAnimal is called using 'super', which refers to the parent class's method. However, removeAnimal is an instance method, and calling it from a static context (like within the static alive array) might lead to unexpected behavior. Consider ensuring that removeAnimal is called on an instance of the Animal class.